index.js

```diff
@@ -71,6 +71,16 @@ const factory = options => {
     opts.issWhitelist.indexOf(token.payload.iss) > -1 ||
     reject(new IssWhitelistError(`iss '${token.payload.iss}' not in issWhitelist`))
 
+  const checkExp = ({ payload }) => {
+    const { exp } = payload
+    const { ignoreExpiration } = verifyOpts
+    const { TokenExpiredError } = jwt
+    const isExpired = !ignoreExpiration && exp < Date.now() / 1000
+    const error = new TokenExpiredError('jwt expired', new Date(exp * 1000))
```

Comment:
Maybe try not to construct an error if it's possible we won't use it? Some error tracking libs attach to the `Error` constructor.